extensions/amp-analytics/0.1/test/test-visibility-model.js

```diff
@@ -1121,4 +1122,111 @@ describes.sandboxed('VisibilityModel', {}, () => {
       expect(spy).to.be.calledTwice;
     });
   });
+
+  describe('reportWhen', () => {
```

Comment:
Could you leverage the existed tests for `reportReadyPromise` https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/test/test-visibility-model.js#L455
Thanks.

Reply:
Done but I couldn't share much with those tests because:

1. The spec changes between tests
2. I avoided `eventSpy = vh.eventResolver_.resolver` to avoid relying on private variables. I use `onTriggerEvent(eventSpy)` to verify that it's reports are sent only once. Using `eventResolver_` seems fragile to me since the `eventResolver_` is re-instantiated in `reset_` but the event spy is not updated.